src/loki/polygon_search.cc

```diff
@@ -24,6 +26,10 @@ using line_bg_t = bg::model::linestring<vm::PointLL>;
 using ring_bg_t = std::vector<vm::PointLL>;
 using namespace vb::json;
 
+typedef bg::model::d2::point_xy<double> point_type;
+typedef bg::model::polygon<point_type> polygon_type;
+typedef bg::model::linestring<point_type> linestring_t;
```

Comment:
just some explanation on the existing boost geom lines:
`BOOST_GEOMETRY_REGISTER_POINT_2D(vm::PointLL, double, bg::cs::geographic<bg::degree>, first, second)`: registers valhalla's PointLL as geographic point type, so now we can do all bg spatial operations on native types instead of creating bg model points.
`BOOST_GEOMETRY_REGISTER_RING(std::vector<vm::PointLL>)` does the same for a ring (mind windedness!). also means we can't use `std::vector<PointLL>` for linestring model, that's why we had to create a bg linestring model a few lines down.

Reply:
Code is removed now.